Move VST plugin code into main repo

mikeoliphant opened the pull request.

dsharlet commented on LiveSPICEVst/EditorView.xaml.cs:
```diff
@@ -0,0 +1,135 @@
+namespace LiveSPICEVst
+{
+    /// <summary>
+    /// Dummy VST host class to use the the UI Test App
```
Can this class also go in the MockVst app? Or is it needed as a placeholder or something? If that's the case, can you update the comment to clarify?

dsharlet reviewed (commented):
Thank you for sharing this! This mostly looks great, I have just one question about the DummyHostInfo, trying to better understand the structure of the code. 

Tagging #16

mikeoliphant commented:
Yeah, it shouldn't be too hard to move it - it will just take some
refactoring of how the MockVst creates the EditorView. If I update my
branch, will it automatically update the pull request?

On Tue, Oct 27, 2020 at 2:42 PM Dillon <notifications@github.com> wrote:

> *@dsharlet* commented on this pull request.
>
> Thank you for sharing this! This mostly looks great, I have just one
> question about the DummyHostInfo, trying to better understand the structure
> of the code.
>
> Tagging #16 <https://github.com/dsharlet/LiveSPICE/issues/16>
> ------------------------------
>
> In LiveSPICEVst/EditorView.xaml.cs
> <https://github.com/dsharlet/LiveSPICE/pull/81#discussion_r513038821>:
>
> > @@ -0,0 +1,135 @@
>
> +﻿using Circuit;
>
> +using Microsoft.Win32;
>
> +using SharpSoundDevice;
>
> +using System;
>
> +using System.Windows;
>
> +using System.Windows.Controls;
>
> +
>
> +namespace LiveSPICEVst
>
> +{
>
> +    /// <summary>
>
> +    /// Dummy VST host class to use the the UI Test App
>
>
> Can this class also go in the MockVst app? Or is it needed as a
> placeholder or something? If that's the case, can you update the comment to
> clarify?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/dsharlet/LiveSPICE/pull/81#pullrequestreview-518159157>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABTGMDZTQQIQWPIS2H33GQLSM45F5ANCNFSM4TBMUBPQ>
> .
>

dsharlet commented:
I believe so, yes. I already see a new commit re: rebuilding the circuit when buttons change.

mikeoliphant commented:
Yeah - I went ahead and committed on my branch. I fixed the issue I was
having with bypass buttons, and moved the DummyHostInfo class into the
MockVst project.

On Tue, Oct 27, 2020 at 2:59 PM Dillon <notifications@github.com> wrote:

> I believe so, yes. I already see a new commit re: rebuilding the circuit
> when buttons change.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/dsharlet/LiveSPICE/pull/81#issuecomment-717564913>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABTGMD3TCC3SVXGZLKCKXSLSM47DNANCNFSM4TBMUBPQ>
> .
>

dsharlet commented:
Oh nice, I see that now. Thanks for that change, I know it seems minor but it helps me understand the structure of the plugin and the test better.

I think this is ready to merge :) I did see some other things (e.g. error handling in the case a circuit fails to load, etc.) but I figured it's easier to address these things as separate changes, this is still a separate branch for now after all.

mikeoliphant commented:
Yep - it is by no means "finished". A pass at error handling would
definitely be useful.

On Tue, Oct 27, 2020 at 3:07 PM Dillon <notifications@github.com> wrote:

> Oh nice, I see that now. Thanks for that change, I know it seems minor but
> it helps me understand the structure of the plugin and the test better.
>
> I think this is ready to merge :) I did see some other things (e.g. error
> handling in the case a circuit fails to load, etc.) but I figured it's
> easier to address these things as separate changes, this is still a
> separate branch for now after all.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/dsharlet/LiveSPICE/pull/81#issuecomment-717568922>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABTGMD4DVJNMOZ3YEFJ33JLSM5AD5ANCNFSM4TBMUBPQ>
> .
>
